server/src/main/java/io/spine/server/delivery/LiveDeliveryStation.java

```diff
@@ -0,0 +1,152 @@
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import com.google.protobuf.Duration;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A station that delivers those messages which are incoming in a live mode.
+ *
+ * <p>This station dispatches all the messages in {@link InboxMessageStatus#TO_DELIVER TO_DELIVER}
+ * status that are present in the passed conveyor. After the messages are dispatched, they are
+ * marked as {@link InboxMessageStatus#DELIVERED DELIVERED}.
+ *
+ * <p>If the idempotence window is {@linkplain DeliveryBuilder#setIdempotenceWindow(Duration) set
+ * in
+ * the system}, all the delivered messages are set to be kept in their storages for the duration,
+ * corresponding to the width of the window. In this way, they will become usable for the potential
+ * de-duplication.
+ *
+ * <p>Before the dispatching, the messages are de-duplicated, taking into account {@linkplain
+ * Conveyor#recentlyDelivered()}
+ * all known delivered messages. In this process, the messages delivered previously and kept for
+ * longer are taken into account as well. The detected duplicates are marked as such
+ * in the conveyor and are removed from the storage later.
+ *
+ * <p>The dispatched messages are re-ordered chronologically. However, the changes in ordering
+ * are not propagated to the conveyor.
+ *
+ * @see CatchUpStation for the station performing the catch-up
+ */
+final class LiveDeliveryStation extends Station {
+
+    /**
+     * The action to use for the delivery of the messages to their targets.
+     */
+    private final DeliveryAction action;
+
+    /**
+     * The current setting of the idempotence window.
+     *
+     * <p>Is {@code null}, if not set.
+     */
+    private final @Nullable Duration idempotenceWindow;
+
+    /**
+     * Creates a new instance of {@code LiveDeliveryStation} with the action to use for the delivery
+     * and the idempotence window.
+     */
+    LiveDeliveryStation(DeliveryAction action, Duration idempotenceWindow) {
+        super();
+        this.action = action;
+        this.idempotenceWindow = !idempotenceWindow.equals(Duration.getDefaultInstance())
+                                 ? idempotenceWindow
+                                 : null;
+    }
+
+    /**
+     * Dispatches the messages from the conveyor.
+     *
+     * @param conveyor
+     *         the conveyor on which the messages are travelling
+     * @return how many messages were delivered and whether there were any errors during the
+     *         dispatching
+     */
+    @Override
+    public final Result process(Conveyor conveyor) {
```

Comment:
Please split into several helper methods.

Reply:
It's short enough and the station-level documentation should help.
Splitting it into more methods will make it worse.

Reply:
We have a case similar to `CatchUpStation` here. Only the author of the code can understand what's going on in this method and why. The Javadoc for this method does not help much because it says that it dispatches messages. It's understood already by the name of the method `process()` and its parameter `Conveyor`. But how does it do it? Why do we have logic which is _that_ complex? 

I think we need another Method Object here. Please consider.